test/orbit_determination.jl

```diff
@@ -0,0 +1,267 @@
+# This file is part of the NEOs.jl package; MIT licensed
+
+using NEOs
+using Dates
+using PlanetaryEphemeris
+using LinearAlgebra
+using Test
+
+using NEOs: NEOSolution, numberofdays
+
+@testset "Orbit Determination" begin
+    @testset "Gauss Method" begin
+        # Load observations 
+        radec = read_radec_mpc(joinpath("data", "RADEC_2023_DW.dat"))
+        # Parameters
+        params = NEOParameters(abstol = 1e-20, order = 25, parse_eqs = true,
+                               bwdoffset = 0.007, fwdoffset = 0.007)
+    
+        # Orbit Determination
+        sol = orbitdetermination(radec, params)
+    
+        # Values by February 4, 2024
+        
+        # Vector of observations
+        @test length(radec) == 123
+        @test numberofdays(radec) < 21.0
+        # Orbit solution
+        @test isa(sol, NEOSolution{Float64, Float64})
+        # Tracklets
+        @test length(sol.tracklets) == 44
+        @test sol.tracklets[1].radec[1] == radec[1]
+        @test sol.tracklets[end].radec[end] == radec[end]
+        @test issorted(sol.tracklets)
+        # Backward integration
+        @test datetime2days(date(radec[1])) > sol.bwd.t0 + sol.bwd.t[end]
+        @test all( norm.(sol.bwd.x, Inf) .< 2 )
+        @test isempty(sol.t_bwd) && isempty(sol.x_bwd) && isempty(sol.g_bwd)
+        # Forward integration
+        @test datetime2days(date(radec[end])) < sol.fwd.t0 + sol.fwd.t[end]
+        @test all( norm.(sol.fwd.x, Inf) .< 2 )
+        @test isempty(sol.t_fwd) && isempty(sol.x_fwd) && isempty(sol.g_fwd)
+        # Vector of residuals
+        @test length(sol.res) == 123
+        @test iszero(count(outlier.(sol.res)))
+        # Least squares fit
+        @test sol.fit.success
+        @test all( sigmas(sol) .< 1e-4 )
+        @test nrms(sol) < 0.38
+        # Scalig factors
+        @test all(sol.scalings .== 1e-6)
+        # Compatibility with JPL
+        JPL = [-1.100331943890894E+00, 2.077277940539948E-01, 4.202679172087372E-02,
+               -4.735673360137306E-03, -1.062665933519790E-02, -6.016253165957075E-03]
+        @test all(abs.(sol() - JPL) .< 27)
+    end
+
+    @testset "Too Short Arc" begin
+        # Optical astrometry file
+        filename = joinpath("data", "2008_EK68.txt")
+        # Download observations
+        get_radec_mpc("designation" => "2008 EK68", filename)
+        # Load observations 
+        radec = read_radec_mpc(filename)
+        # Delete astrometry file
+        rm(filename)

```

Comment:
EDIT: (deleted my previous suggestion, since it had a typo 😅)

In some of the tests there seems to be a pattern of opening a file, constructing a Vector{RadecMPC{T}} and then deleting the file. I suggest to add a function like this, which can reduce this pattern to a single line

```julia
function fetch_radec_mpc(s::String)
    f = tempname()
    get_radec_mpc("designation" => s, f)
    radec = read_radec_mpc(filename)
    rm(f)
    return radec
end
```

Reply:
Thanks for the suggestion, I've added `fetch_radec_mpc` to `src/observations/radec_mpc.jl` and modified the tests accordingly.